state_synchronizer/src/coordinator.rs

```diff
@@ -303,7 +309,11 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
 
     /// processes batch of transactions downloaded from peer
     /// executes transactions, updates progress state, calls callback if some sync is finished
-    async fn process_chunk_response(&mut self, mut response: GetChunkResponse) -> Result<()> {
+    async fn process_chunk_response(
+        &mut self,
+        peer_id: &PeerId,
+        mut response: GetChunkResponse,
+    ) -> Result<()> {
         let txn_list_with_proof = response.take_txn_list_with_proof();
 
         if let Some(version) = txn_list_with_proof
```

Comment:
we need to go more carefully with cases. Cause, for instance, if version is None here, chunk is still invalid